### ISSUE TITLE
MQTT Bridge - Allow prefix/suffix to remap of topics

### ISSUE THREAD

**Issue by aitorroca** [nanomq/nanomq#1746]
**Is your feature request related to a problem? Please describe.**
Configuring a bridge, if you try to rewrite the topic with a wildcard:

```
forwards = [
  {
    remote_topic = "device_id/topic1/#"
    local_topic = "topic1/#"
  },
]
```

It shows:

`2024-04-13 16:47:23 [1] ERROR /home/runner/work/nanomq/nanomq/nng/src/supplemental/nanolib/conf_ver2.c:970 conf_bridge_node_parse: No wildcard +/# should be contained in remote topic in forward rules.`

Similar issues: #1684, #1463, https://github.com/nanomq/nanomq/issues/1174#issuecomment-1769265904

**Describe the solution you'd like**
Similar to what eclipse mosquitto allows: "_The local-prefix and remote-prefix options allow topics to be remapped when publishing to and receiving from remote brokers. This allows a topic tree from the local broker to be inserted into the topic tree of the remote broker at an appropriate place._" (https://mosquitto.org/man/mosquitto-conf-5.html)

As example, in mosquitto, that configuration line allows to remap (adding a prefix in the remote topic) with a wildcard:

`topic topic1/# both 0 "" device_id/`

**Describe alternatives you've considered**
Right now, the alternative I've tested is:

{
remote_topic = ""
local_topic = "device_id /topic1/#"
}

That works, but it doesn't allow a topic to be remapped with a wildcard, moving more work to the local configuration part.

**Comment by RanMaoyi** [comment]
Thank you for taking the time to share your suggestion with us. Your proposed enhancement is indeed interesting and aligns well with our goals for further development.  We will carefully consider it as we plan our roadmap for future updates.

**Comment by JaylinYu** [comment]
Nope, MQTT forbidden you to publish msg to wildcard. simply cannot.
What you really are asking for is a prefix/suffix of the remote topic that allows modifying topic tree in the remote broker. the way mosquito implement it make it looks like you are publishing msg to wildcard.

**Comment by JaylinYu** [comment]
Now it is done

**Comment by aitorroca** [comment]
First of all, I want to express my gratitude to the development team for your continuous efforts in improving the broker. Your dedication is greatly appreciated.

**Issue Details:**

I understand that the prefix and suffix functionality has been implemented for subscriptions, as per the documentation

  - **suffix**: A suffix string that will be added to the local topic (added to the original topic if you leave `local_topic` as null).
  - **prefix**: A prefix string that will be added to the local topic (added to the original topic if you leave `local_topic` as null).

However, despite the presence of the prefix and suffix options, they don't seem to address our specific use case.

**Our Use Case:**

We are attempting to configure a subscription where we receive messages from a remote topic that includes a UUID prefix, and we want to map these messages to a local topic without the UUID.

For example, we have dynamic subtopics under a remote topic structured as `785fa9a4-1382-4c58-a11e-c89d94569f6b/zb/#`. We wish to receive these messages under the local topic `zb/#`, effectively stripping the UUID prefix.

Our initial configuration attempt was:

```hocon
subscription = [ 
  {
    remote_topic = "785fa9a4-1382-4c58-a11e-c89d94569f6b/zb/#"
    local_topic = "zb/#"
    qos = 1
  },
]
```

This resulted in the error:

`2024-11-04 22:37:58 [1] ERROR /home/runner/work/nanomq/nanomq/nng/src/supplemental/nanolib/conf_ver2.c:1267 conf_bridge_node_parse: No wildcard +/# should be contained in local topic in subscription rules`

Alternatively, we tried:

```hocon
subscription = [ 
  {
    remote_topic = "785fa9a4-1382-4c58-a11e-c89d94569f6b/zb/#"
    local_topic = ""
    qos = 1
  },
]
```

But this configuration results in messages being received with the UUID prefix included in the local topic, which we want to avoid.

Using the prefix and suffix options doesn’t seem to help in this scenario because they only allow us to add strings to the local topic, not to modify or remove parts of it.

**Feature Request:**

Would it be possible to enhance the subscription functionality to allow for topic remapping that includes the ability to strip or replace parts of the topic string? Specifically, we are looking for a way to map a remote topic with a known prefix to a local topic without that prefix, while retaining the dynamic subtopics.

For example, implementing a way to specify that a certain part of the remote topic should be removed in the local topic, perhaps through a parameter like strip_prefix or by supporting variable substitution in the topic strings.

This feature would allow configurations such as:

```hocon
subscription = [ 
  {
    remote_topic = "785fa9a4-1382-4c58-a11e-c89d94569f6b/zb/#"
    local_topic = "zb/#"
    qos = 1
  },
]
```

To further illustrate, suppose we have messages published to a multi-level subtopic like:

- 785fa9a4-1382-4c58-a11e-c89d94569f6b/zb/device1/status
- 785fa9a4-1382-4c58-a11e-c89d94569f6b/zb/device2/config

With the requested feature, we aim for these messages to be mapped to:

- zb/device1/status
- zb/device2/config

This would allow us to handle subscriptions to multi-level subtopics dynamically, without needing to explicitly define each subtopic.

Additionally, I want to mention that the forwarding functionality is working perfectly, and we greatly appreciate that improvement.

Once again, thank you for your hard work. This enhancement would greatly benefit those of us dealing with dynamic subtopics and needing more flexible topic remapping capabilities in subscriptions. Your consideration of this feature request is highly appreciated.
